scripts/components/Tooltip/utils.js

```diff
@@ -0,0 +1,52 @@
+export default {
+  /**
+   * Calculates the position of the overlay (tooltip)
+   *
+   * The math here is as simple as the requirement. We just
+   * want the tooltip to appear in the center of the provided
+   * placement (for example, horizontally center it if top or btm;
+   * or vertically center it if left or right).
+   *
+   * To center `top`:
+   * (left): (<A> + <B / 2> + <C / 2>)
+   * A = offsetLeft; distance between the left-most screen and `trigger`
+   * B = ./.
+   * C = ./.
+   *
+   * I left a comment here because the terminologies were a bit confusing.
+   * There's duplicated code (for example, top and bottom had the same `left` offset;
+   * left and right had the same `top` offset) because obviously, it would be unnecessarily
+   * more expensive and/or the code with be more lengthy (which is seriously fucking bad).
+   */
+  calculatePosition(trigger, overlay, placement) {
+    const coords = trigger.getBoundingClientRect();
+    const ADD_MARGIN = 8;
```

Comment:
Like I said, I think putting a margin to the class itself would be better than adding it to the computations. You should ask @JefMari here as he worked on the dropdown.